CDCgov/cdc-open-viz
# Dev 9346 nested dropdown not populating filter options viz level

## [Replace With Ticket Number]



## Testing Steps

Filter is not populating with the filter options after the group and subgroup settings are selected. 

Query Parameters checkbox is not working.

Scenario: In Editor, user has selected Bar Chart Visualization.

1. Import data. The following link is a csv file with data that works for this example. Download and upload the attached file:  nested_dropdown_sample.csv. Click Vertical, No to multiple series, and then Configure your Visualization button.

2. In Visualization, click on Data Series accordion, and select Data_Value in the Data Series dropdown.

3. Click on the Date/Category Axis accordion. And Select YearStart from the DATA KEY dropdown

4. Click the I'm Done button.

5. Click the Filters accordion and Add Filter

6. Choose Nested Dropdown from the FILTER STYLE dropdown

7. In the LABEL text box write "Stratification 1"

Expected: Label in the right screen should say "Stratification 1"

8. In the FILTER GROUPING dropdown, select StratificationCategory1

9. In the FILTER SUBGROUPING dropdown, select Stratification1

Expected: There should be a nested dropdown filter which is showing a option but, there are no other options to be selected. 

Workaround: After step 9, click on Group Order and select Ascending, Descending, or Custom order. This will make the Filter Populate with the all of the options. 

10. click the CREATE QUERY PARAMETERS checkbox.

Expected: There is no change or selections popping up. 

## Self Review

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

## Screenshots (if applicable)



## Additional Notes
